File: src/main/java/io/github/shirohoo/library/domain/user/UserLoanHistory.java
package io.github.shirohoo.library.domain.user;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

@Entity
public class UserLoanHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private User user;

    private String bookTitle;

    private boolean isReturn;

    public UserLoanHistory() {
    }

    public UserLoanHistory(User user, String bookTitle, boolean isReturn) {
        this.user = user;
        this.bookTitle = bookTitle;
        this.isReturn = isReturn;
    }

    public String getBookTitle() {
        return this.bookTitle;
    }

    public void doReturn() {
        this.isReturn = true;
    }
}

File: src/main/kotlin/io/github/shirohoo/library/domain/user/UserLoanHistory.kt
package io.github.shirohoo.library.domain.user

import javax.persistence.*

@Entity
class UserLoanHistory(
    @ManyToOne
    val user: User,

    val bookTitle: String,

    status: UserLoanStatus,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
) {
    var status: UserLoanStatus = status
        private set

    val isReturn: Boolean = this.status == UserLoanStatus.RETURNED

    fun doReturn() {
        this.status = UserLoanStatus.RETURNED
    }
}